lib/spack/env/cc

```diff
@@ -325,6 +343,8 @@ while [ -n "$1" ]; do
                     die "-Wl,-rpath was not followed by -Wl,*"
                 fi
                 rp="${arg#-Wl,}"
+            elif [[ "$arg" = --enable-new-dtags ]] ; then
```

Comment:
Reading `bash` is not my best skill, but I think we are currently not treating NAG the same as the other compilers:
- RPATHs are not collected in `rp` as there's no condition that matches `-Wl,-Wl,,-rpath,,XXX`
- Every `-Wl,-Wl,,-rpath,,XXX` will be appended to `other_args`

Am I misreading something? In case: do we want to fix this here or, having the issue a minimal impact, can we defer it to a later PR?

Reply:
I wish I understood the NAG linker argument syntax there -- I wouldn't mind actually fixing this here if you're up for it.  I just don't have a NAG compiler to test on.  I looked at the NAG manual and it doesn't *seem* to indicate that we need two `-Wl,`'s.  Do you know who we could ask?  I suspect we don't have many NAG users.

Reply:
NAG calls gcc for linking, which calls ld. Therefore, when we specify a flag with a single prefix -Wl for NAG, it will pass it to gcc, which won't understand an ld-specific flag. So, we have to wrap an ld-specific flag twice: the first level will be unwrapped by NAG, the second one - by gcc.